### ISSUE TITLE
Incorrect indentation check for method preceded by annotation, with method parameter on separate line

### ISSUE THREAD

**Issue by andersthorbeck** [checkstyle/checkstyle#5154]
Anderss-MacBook-Pro:checkstyle andtho$ javac Test.java 

Anderss-MacBook-Pro:checkstyle andtho$ cat Test.java 
```java
package bugreport;

class Test {

  @interface Annotation1 {
    String field1();
    String field2();
  }

  @interface Annotation2 {};

  @Annotation1(field1 = "foo", field2 = "bar")
  public @Annotation2 String method( // violation
      String param
  ) {
    return null;
  }

}
```

Anderss-MacBook-Pro:checkstyle andtho$ cat config.xml
```xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
        "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
        "http://checkstyle.sourceforge.net/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="Indentation">
      <property name="basicOffset" value="2"/>
      <property name="braceAdjustment" value="0"/>
      <property name="caseIndent" value="2"/>
      <property name="throwsIndent" value="4"/>
      <property name="lineWrappingIndentation" value="4"/>
      <property name="arrayInitIndent" value="2"/>
    </module>
  </module>
</module>
```

Anderss-MacBook-Pro:checkstyle andtho$ java -jar checkstyle-8.2-all.jar -c config.xml Test.java
```
Starting audit...
[ERROR] /Users/andtho/code/sandbox/bug_reports/checkstyle/Test.java:13: 'public' has incorrect indentation level 2, expected level should be 6. [Indentation]
Audit done.
Checkstyle ends with 1 errors.
```

---------------

Expected no indentation error.

The error occurs when the parameter `param` is on a new line; it does not occur when the entire method signature (except `@Annotation1`) is on one line. It also does not occur when `@Annotation1` is omitted.

I need to be able to put parameters on a new line, as the method signature exceeds the maximum line width (in real-life code), so this is an actual issue.

Note, when I indent `public` to indentation level 6, the expected indentation level is increased to 10, and then to 14, 18, etc.

**Comment by AADudkin** [comment]
I'm on it

**Comment by romani** [comment]
@AADudkin , Indentation Check is the most complicated Check in whole Checkstyle, and probably requires complete rewrite. Try to fix, but feel free to drop this issue. There are more issues that is more ease to resolve and they are more painful for users.

**Comment by AADudkin** [comment]
Dropped issue. Currently don't have enough time to contribute.

**Comment by djydewang** [comment]
I am on it.

**Comment by rnveach** [comment]
Fix was merged
